stream_alert/classifier/parsers.py

```diff
@@ -0,0 +1,808 @@
+            # value, so we should try to parse it as one and use the result
+            if isinstance(schema[key], dict):
+                parsed_data = self._extract_records([row[index]], schema[key])
+                if parsed_data and parsed_data[0][1] is True:
+                    parsed_payload[key] = parsed_data[0][0]
+                    continue
+                else:
+                    return False  # break the loop if the nested data was invalid
+
+            # extract the keys from the row via the index
+            parsed_payload[key] = row[index]
+
+        return parsed_payload
+
+
+@parser
+class KVParser(ParserBase):
+    """Key/value record parser"""
+    _type = 'kv'
+
+    @property
+    def delimiter(self):
+        # default delimiter = ' '
+        return str(self.configuration.get('delimiter', ' '))
+
+    @property
+    def separator(self):
+        # default separator = '='
+        return str(self.configuration.get('separator', '='))
+
+    def _parse(self, data):
+        """Parse a key/value string into a dictionary
+
+        Args:
+            data (str): Data to be parsed
+
+        Returns:
+            list<tuple>: List of tuples with records and their parsing status
+                Example: [({'key': 'value'}, True)]
+        """
+        record = self._extract_record(data)
+        return [(record, True)] if record else [(data, False)]
+
+    def _extract_record(self, data):
+        """Extract the key/value record from the string of data
+
+        Args:
+            data (str): String of data from which to extract key/value information
+
+        Returns:
+            dict: Parsed keys and values from the string data, False upon failure
+        """
+        kv_payload = {}
+        try:
+            # remove any blank strings that may exist in our list
+            fields = [field for field in data.split(self.delimiter) if field]
+            # first check the field length matches our # of keys
+            if len(fields) != len(self.schema):
+                return False
+
+            for index, field in enumerate(fields):
+                # verify our fields contains the separator
+                if self.separator not in field:
+                    LOGGER.error('separator \'%s\' not found in field: %s', self.separator, field)
+                    continue
+
+                # only take data preceeding the first occurence of the field as the key
+                key, value = field.split(self.separator, 1)
+                # handle duplicate keys
+                if key in kv_payload:
+                    # load key from our configuration
+                    kv_payload[self.schema.keys()[index]] = value
+                else:
+                    # add the data value
+                    kv_payload[key] = value
+
+        except UnicodeDecodeError:
+            return False
+
+        return kv_payload
+
+
+@parser
+class SyslogParser(ParserBase):
```

Comment:
while we're factoring things, could this be a subclass of a more generic RegexParser?

Reply:
this is a good idea. I'll try to address now but may come in a later PR